src/tools/crawler/crawl.rs

```diff
@@ -0,0 +1,121 @@
+//! Contains structs and methods to crawl ripple network according to instruction at https://xrpl.org/peer-crawler.html
+
+use std::{
+    net::{IpAddr, SocketAddr},
+    num::ParseIntError,
+    time::Duration,
+};
+
+use reqwest::{Client, StatusCode};
+use serde::Deserialize;
+use thiserror::Error;
+use tokio::time::Instant;
+
+const CRAWLER_DEFAULT_PORT: u16 = 51235;
+const CRAWLER_TIMEOUT: Duration = Duration::from_secs(10);
+
+#[allow(dead_code)]
```

Comment:
is the whole crawler module not feature-gated?

Reply:
It is. This `allow(dead_code)` is about (yet) unused fields in `Peer`